fhirflat/resources/base.py

```diff
@@ -201,58 +259,45 @@ def ingest_to_flat(cls, data: pd.DataFrame, filename: str) -> pd.DataFrame | Non
         Returns
         -------
         pd.DataFrame or None
-            A dataframe containing the flat_dict and validation errors.
+            A dataframe containing the FHIRflat data.
         """
 
         data.loc[:, "flat_dict"] = cls.ingest_backbone_elements(data["flat_dict"])
 
-        # Creates a columns of FHIR resource instances
-        data["fhir"] = data["flat_dict"].apply(lambda x: cls.create_fhir_resource(x))
-
-        validation_error_mask = data["fhir"].apply(
-            lambda x: isinstance(x, ValidationError)
-        )
-
-        valid_fhir = data[~validation_error_mask].copy()
-
-        # flattens resources back out
-        flat_df = valid_fhir["fhir"].apply(lambda x: x.to_flat())
+        flat_df = pd.json_normalize(data["flat_dict"])
 
         if not flat_df.empty:
-            # create FHIR expected date format
-            for date_cols in [
-                x
-                for x in flat_df.columns
-                if ("date" in x.lower() or "period" in x.lower() or "time" in x.lower())
-            ]:
-                # replace nan with None
-                flat_df[date_cols] = flat_df[date_cols].replace(np.nan, None)
-
-                # convert datetime objects to ISO strings
-                # (stops unwanted parquet conversions)
-                # but skips over extensions that have floats/strings rather than dates
-                flat_df[date_cols] = flat_df[date_cols].apply(
-                    lambda x: (
-                        x.isoformat()
-                        if isinstance(x, datetime.datetime)
-                        or isinstance(x, datetime.date)
-                        else x
-                    )
+            # apply the coding column formatting in here
+            system_columns = flat_df.columns[flat_df.columns.str.endswith(".system")]
+            for coding_col in system_columns:
+                col = coding_col.removesuffix(".system")
+                flat_df = flat_df.apply(
+                    lambda x, c=col: util.condense_codes(x, c), axis=1
                 )
-
-            for coding_column in [
+            flat_df.drop(columns=system_columns, inplace=True)
+
+            potential_dense_cols = [
+                x for x in cls.backbone_elements.keys() if x in flat_df.columns
+            ]
+            list_lengths = [
+                len(flat_df[x].dropna().iloc[0]) for x in potential_dense_cols
```

Comment:
Are all these columns list-like? Otherwise might give length of a string or give an error if there is no `__len__` defined

Reply:
It should be, but I'll add a check